### ISSUE TITLE
Better representation of typed arrays in the browser console

### ISSUE THREAD

**Issue by janpaul123** [Zaplib/zaplib#98]
@disambiguator "Would be cool if we could clean up our monkey patching such that normal TypedArrays don't get represented as ZapTypedArrays, not sure how though. i.e."

![image](https://user-images.githubusercontent.com/177461/155807769-ef4cc2f2-42d1-4b64-a623-eee76e12eae8.png)

Maybe this still works? https://stackoverflow.com/a/38652208

Or maybe we can create it not as an ES6 class but in the old prototype style so we can give it a custom name?

@disambiguator: "I dont want to try right this moment but also curious what would happen if we did.."

```
diff --git a/zaplib/web/zap_buffer.ts b/zaplib/web/zap_buffer.ts
index 9f5ce3f..b0a54b0 100644
--- a/zaplib/web/zap_buffer.ts
+++ b/zaplib/web/zap_buffer.ts
@@ -81,7 +81,7 @@ function zapBufferExtends(cls: any) {
         super(...args);
         this.__zaplibBuffer = buffer;
       } else {
-        super(...args);
+        cls(...args);
       }
     }
```

**Comment by janpaul123** [comment]
Right but that makes sense right? In that else clause it's not backed by a `ZapBuffer` so it isn't a zap array any more? So maybe calls to `getZapParamType` should be preceded with `isZapBuffer` (aren't they already?) since that should return false in that case.

**Comment by disambiguator** [comment]
Not necessarily. getZapParamType is also used by `createBuffer` / `createReadOnlyBuffer` to determine what kind of param should be created from a normal TypedArray.

**Comment by janpaul123** [comment]
Oh I see I understand now. Hm, that is strange. I wonder what happens if we use a traditional constructor, not an ES6 class. I suspect that that will work with `return cls(...args)`.
